libs/@local/harpc/net/src/session/server/config.rs

```diff
@@ -0,0 +1,95 @@
+use core::{num::NonZero, time::Duration};
+
+use tokio::sync::Semaphore;
+
+use crate::macros::non_zero;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ConcurrentConnectionLimit(u32);
+
+impl ConcurrentConnectionLimit {
+    pub fn new(value: u32) -> Option<Self> {
```

Comment:
`0` literally means that we won't accept any connections (this actually works) I wanted to avoid a panic at `MAX_PERMITS` which is why I added this (`tokio` `Semaphore` panics if it is too high)